sklearn/cluster/_spectral.py

```diff
@@ -259,9 +259,26 @@ def spectral_clustering(
         consecutive runs in terms of inertia. Only used if
         ``assign_labels='kmeans'``.
 
-    eigen_tol : float, default=0.0
-        Stopping criterion for eigendecomposition of the Laplacian matrix
-        when using arpack eigen_solver.
+    eigen_tol : float, default="auto"
+        Stopping criterion for eigendecomposition of the Laplacian matrix.
+        If `eigen_tol="auto"` then the passed tolerance will depend on the
+        `eigen_solver`:
+
+        - If `eigen_solver="arpack"`, then `eigen_tol=0.0`;
+        - If `eigen_solver="lobpcg"` or `eigen_solver="amg"`, then
+          `eigen_tol=None` which allows the underlying SciPy solver to
+          automaticaly resolve the value according to their heuristics.
+
+          .. note::
+            For details on the heuristics used by SciPy, please see:
+            https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.linalg.lobpcg.html
```

Comment:
We can link directly to the SciPy function as follows:

```suggestion
          `eigen_tol=None` which configures the underlying `lobpcg` solver to
          automatically resolve the value according to their heuristics. See,
          :func:`scipy.sparse.linalg.lobpcg` for details.
```

In both amg and lobpcg, the `eigen_tol` parameter configures a `lobpcg` call.

This comment applies to the other docstrings.